app/src/main/java/org/oppia/android/app/profile/ResetPinDialogFragmentPresenter.kt

```diff
@@ -45,20 +46,22 @@ class ResetPinDialogFragmentPresenter @Inject constructor(
       viewModel = resetViewModel
     }
 
-    binding.inputPin.setInput(resetViewModel.inputPin.get().toString())
-    binding.inputPin.addTextChangedListener(object : TextWatcher {
-      override fun onTextChanged(confirmPin: CharSequence?, start: Int, before: Int, count: Int) {
-        confirmPin?.let {
-          resetViewModel.inputPin.set(confirmPin.toString())
+    // [onTextChanged] is a extension function defined at [TextInputEditTextHelper]
+    binding.resetPinInputPinEditText.onTextChanged { confirmPin ->
+      confirmPin?.let {
+        if (
+          resetViewModel.errorMessage.get()?.isNotEmpty()!! && resetViewModel.inputPin.get() == it
+        ) {
+          resetViewModel.inputPin.set(it)
+        } else {
+          resetViewModel.inputPin.set(it)
           resetViewModel.errorMessage.set("")
         }
       }
+    }
 
-      override fun afterTextChanged(confirmPin: Editable?) {}
-      override fun beforeTextChanged(p0: CharSequence?, start: Int, count: Int, after: Int) {}
-    })
-
-    binding.inputPin.setLabel("$name's New PIN")
+    binding.resetPinInputPin.hint = activity.resources
```

Comment:
please put a comment here that why we are setting string here and what are the reasons of not setting this in the XML as it will prevent to shift to XML file.

Reply:
Done